mono/metadata/metadata.c

```diff
@@ -5435,6 +5440,31 @@ do_mono_metadata_type_equal (MonoType *t1, MonoType *t2, gboolean signature_only
 	if (t1->type != t2->type || t1->byref != t2->byref)
 		return FALSE;
 
+	if (t1->has_cmods != t2->has_cmods)
+		return FALSE;
+
+	if (t1->has_cmods) {
+		MonoCustomModContainer *cm1 = mono_type_get_cmods (t1);
+		MonoCustomModContainer *cm2 = mono_type_get_cmods (t2);
+
+		g_assert (cm1);
+		g_assert (cm2);
+
+		if (cm1->count != cm2->count)
+			return FALSE;
+
+		for (int i=0; i < cm1->count; i++) {
+			ERROR_DECL (error);
+			MonoClass *c1 = mono_class_get_checked (cm1->image, cm1->modifiers [i].token, error);
+			mono_error_assert_ok (error);
+			MonoClass *c2 = mono_class_get_checked (cm2->image, cm2->modifiers [i].token, error);
+			mono_error_assert_ok (error);
+
+			if (c1 != c2)
+				return FALSE;
```

Comment:
Does custom modifier order matter? if I have `t modreq(mod1) modreq(mod2) t`  vs `t modreq(mod2) modreq(mod1)` are those the same type or different?